app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,56 @@
+class MerchantsController < ApplicationController
+  before_action :require_user, only: [:edit, :show]
+  before_action :redirect_if_logged_in, only: [:new, :create]
+
+  def index
+  end
+
+  def new
+    @merchant = Merchant.new
+  end
+
+  def create
+  @merchant = Merchant.create(merchant_params)
+    if @merchant.save
+# makes it so they don't have to login after they sign up. Takes the session data so it can run the redirect.
+      session[:merchant_id] = @merchant.id
+      redirect_to merchant_path(@merchant)
+    else
+      flash[:error] = "Try again. Or you may already be a user. Try logging in."
+      render :new
+    end
+  end
+
+  def show
+    id = @current_user.id
+    @merchant = Merchant.find(id)
+    @products = @merchant.products
+    status = params[:sort]
+
+    if status == "all" || status == nil
+      @orders = @merchant.orders
+    else
+      @orders = @merchant.orders.where(status: status)
+    end
+    @all_orders = @merchant.orders
+    @pending_revenue ||= 0
```

Comment:
I have some logic in the merchant show view that totaled up the orders by status.  I now realize that the logic should be in the show method in the controller.  The ||=0 was to keep the totalling logic from returning nil. 

Line 30 is using | | to set a default before the user attempts to filter their orders by status.  Is this an ok way to write that @Hamled ?  Is there a way to make it more readable, or should there be comments?